src/engine/bpmcontrol.cpp

```diff
@@ -219,6 +219,9 @@ void BpmControl::slotTapFilter(double averageLength, int numSamples) {
     if (numSamples < 4)
         return;
 
+    if (!m_pBeats)
```

Comment:
Nice catch! Could you grab a reference to the shared pointer first so that it can't be deleted between this line and the setBpm call below?

```c++
auto pBeats = m_pBeats;
if (!pBeats) {
  return;
}
...
pBeats->setBpm(averageBpm);
```

Reply:
I can do that. Other usages in this file don't do that though. If that's necessary, there might be some race conditions.